### ISSUE TITLE
Synchronise external functions from Octave to their most recent versions

### ISSUE THREAD

**Issue by gllmflndn** [fieldtrip/fieldtrip#1941]
Hello,

We [recently noticed](https://github.com/translationalneuromodeling/tapas/issues/166) that some of the functions in [external/signal](https://github.com/fieldtrip/fieldtrip/tree/master/external/signal) that were borrowed from the Octave Signal toolbox are not up to date  - and crucially not compatible with MATLAB versions. One example is [blackmanharris](https://github.com/fieldtrip/fieldtrip/blob/master/external/signal/blackmanharris.m) which was [updated in Octave in 2013 to return a column vector](https://sourceforge.net/p/octave/signal/ci/573a10d508e78a6178c811be5c11b67d44f04aac/) but still returns a row vector in FieldTrip.
I haven't checked how many of them (in the Signal but also the [Statistics toolbox](https://github.com/fieldtrip/fieldtrip/tree/master/external/stats)) would need to be updated. I am not sure how much it would affect FieldTrip if the behaviour of some of the functions was changed after being updated.

Many thanks.

**Comment by schoffelen** [comment]
Thanks Guillaume, we'll check it out. It's been a while that I used the Blackmanharris window :).

**Comment by robertoostenveld** [comment]
I propose that we write `test_compat_signal`, `test_compat_images`, and `test_compat_stats`. These should test the known compat functions, and also give an error if new functions are being added in the future without them being part of the test script.

**Comment by schoffelen** [comment]
OK, I strarted process of fixing blackmanharris, and writing a test function for the test_external_signal

So far, I have updated the octave code : I checked the currently supported functions, and updated them a bit if the sourceforge code looked more modern than the current version in the FT repo. 

I have started a test function, that for now only performs some internal sanity checks, which I borrowed from the octave code.

To do:
-finalize the sanity check code
-build an explicit comparison between the matlab implementation and the external computations
-do the same for the stats toolbox.
